dmn-engine/src/main/scala/org/camunda/dmn/DmnEngine.scala

```diff
@@ -51,6 +51,18 @@ object DmnEngine {
       }
     }
 
+    def audit(result: Either[Failure, Val],
+              decisionLogic: ParsedDecisionLogic,
+              funct: Val => EvaluationResult): Either[Failure, Val] = {
+      result match {
+        case Right(result) =>
+          audit(decisionLogic, funct(result))
+        case Left(failure) =>
+          audit(decisionLogic, funct(ValError(failure.message)))
+      }
+      result
+    }
```

Comment:
* Switch parameters for readability
* No return type. First, it seems a bit odd that an audit function returns something. Second, chaining the audit call seems handly but it makes it harder to read compared to a more imperative style. See the comments where it is used. 
* Default factory for simple cases.
 
```suggestion
    def audit(decisionLogic: ParsedDecisionLogic,
              result: Either[Failure, Val],
              resultFactory: Val => EvaluationResult  = SingleEvaluationResult) {
      result match {
        case Right(result) =>
          audit(decisionLogic, resultFactory(result))
        case Left(failure) =>
          audit(decisionLogic, resultFactory(ValError(failure.message)))
      }
      result
    }
```